core/domain/user_jobs_one_off.py

```diff
@@ -761,3 +761,43 @@ def map(model):
     @staticmethod
     def reduce(key, values):
         yield (key, len(values))
+
+
+class UniqueHashedNormalizedUsernameAuditJob(
+        jobs.BaseMapReduceOneOffJobManager):
+    """Job that checks that the hashed normalized usernames are unique."""
+
+    @classmethod
+    def enqueue(cls, job_id, additional_job_params=None):
+        # We can raise the number of shards for this job, since it goes only
+        # over one type of entity class.
+        super(UniqueHashedNormalizedUsernameAuditJob, cls).enqueue(
+            job_id, shard_count=32)
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [user_models.UserSettingsModel]
+
+    @staticmethod
+    def map(model):
+        if model.normalized_username is None:
+            yield ('SUCCESS USERNAME NONE', 1)
+        else:
+            yield (
+                'SUCCESS',
```

Comment:
I suggest making use of the MapReduce structure and keying by the hash. I.e. yield (hash, count). In reduce(), only yield if count is not equal to 1.

(Or perhaps, if we need debug info, yield (hash, usernames_list) and only yield in reduce() if len(usernames_list) != 1.)

Reply:
Done.